File: Leetcode-Problems/Find Lucky Integer in an Array.cpp
class Solution
{
public:
    int findLucky(vector<int> &arr)
    {
        unordered_map<int, int> mp;
        for (int i : arr)
            mp[i]++;
        int ans = -1;
        for (auto i : mp)
        {
            if (i.first == i.second)
            {
                ans = max(ans, i.second);
            }
        }
        return ans;
    }
};

File: Leetcode-Problems/Find Lucky Integer in an Array.java
class Solution {
    public int findLucky(int[] arr) {
        int[] freq = new int[501];
        for (int i = 0; i < arr.length; i++) {
            freq[arr[i]]++;
        }
        int maxi = -1;
        for (int i = 1; i < 501; i++) {
            if (freq[i] == i) {
                maxi = Math.max(maxi, i);
            }
        }
        return maxi;
    }
}